rust/core-lib/src/watcher.rs

```diff
@@ -213,22 +213,35 @@ impl FileWatcher {
     }
 
     /// Takes ownership of this `Watcher`'s current event queue.
-    pub fn take_events(&mut self) -> VecDeque<(WatchToken, DebouncedEvent)> {
+    pub fn take_events(&mut self) -> VecDeque<(WatchToken, Event)> {
         let mut state = self.state.lock().unwrap();
         let WatcherState { ref mut events, .. } = *state;
         mem::replace(events, VecDeque::new())
     }
 }
 
 impl Watchee {
-    fn wants_event(&self, event: &DebouncedEvent) -> bool {
-        use self::DebouncedEvent::*;
-        match *event {
-            NoticeWrite(ref p) | NoticeRemove(ref p) | Create(ref p) | Write(ref p)
-            | Chmod(ref p) | Remove(ref p) => self.applies_to_path(p),
-            Rename(ref p1, ref p2) => self.applies_to_path(p1) || self.applies_to_path(p2),
-            Rescan => false,
-            Error(_, ref opt_p) => opt_p.as_ref().map(|p| self.applies_to_path(p)).unwrap_or(false),
+    fn wants_event(&self, event: &Event) -> bool {
+        match &event.kind {
+            EventKind::Create(CreateKind::Any)
+            | EventKind::Remove(RemoveKind::Any)
+            | EventKind::Modify(ModifyKind::Any)
+            | EventKind::Modify(ModifyKind::Metadata(MetadataKind::Any)) => {
+                if event.paths.len() == 1 {
+                    self.applies_to_path(&event.paths[0])
+                } else {
+                    false
```

Comment:
This basically says "if path length is not 1 ...I dont want this event".

Reply:
Wouldn't
```
if let Some(path) = event.paths.get(0) {
    self.applies_to_path(path)
}
```

Be clearer?

Reply:
It is but not here. What if len is more than 1? Should we still get the first element? We cannot deterministically say first element is THE element we are looking for.

Reply:
The 'rustic' way to write this code, I think, would be:

```rust
event.paths.first().map(|p| self.applies_to_path(p)).unwrap_or(false)
```

Reply:
Hm, that doesn't address @sjoshid 's concern if len is more than 1, does it?

Reply:
yeah. @cmyr I commented in other conversation.